src/wrapper/objects/auto_byte_array.rs

```diff
@@ -58,17 +58,45 @@ impl<'a, 'b> AutoByteArray<'a, 'b> {
         self.ptr.as_ptr()
     }
 
-    /// Commits the result of the array, if it is a copy
+    fn commit_byte_array_elements(&mut self) -> Result<()> {
```

Comment:
Nit: I'd place these utility methods after their callers, so that it is easier to see how they are used.